source/_posts/2015-03-23-artsy-technology-stack-2015.markdown

```diff
@@ -0,0 +1,52 @@
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/developers.png'></center>
+
+Artwork, artist and other metadata is uploaded by our partners via a home-grown content-management system built entirely from scratch in 2014 on Ruby-on-Rails. This was a deliberate "boring" choice that continues to serve us well. We have adopted a common model for admin-type apps with a shared component library and a reusable UI, all implemented as Rails engines. Using these components we are able to quickly and easily compose applications to manage fairs or auctions, for example.
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/cms.png'></center>
+
+Our family of mobile applications includes [Artsy for iOS](http://iphone.artsy.net), which is a hybrid app written in Objective-C, and a [bidding kiosk](https://github.com/artsy/eidolon), written in Swift. Both are open-source [here](https://github.com/artsy/eigen) and [here](https://github.com/artsy/eidolon).
+
+<center><img src='/images/2015-03-23-artsy-technology-stack-2015/folio.jpg'></center>
+
+A lot of data, including the artwork Similarity graph that powers the Art Genome Project, is processed offline by a generic job engine, written in Ruby. We take data snapshots from MongoDB, run jobs on the data and export data back to the database. Other recently rewritten services include image processing, which creates thumbnails, image tiles for deep zoom and wtermarks high quality JPEGs. Several new applications use [PosgreSQL](http://www.postgresql.org). We also started doing Analytics with [Snowplow](https://github.com/snowplow/snowplow) and data warehousing in [AWS Redshift](http://aws.amazon.com/redshift/). We still do love [Graphite](http://graphite.wikidot.com) and [Statsd](https://github.com/etsy/statsd). Smaller systems usually start on [Heroku](https://dashboard.heroku.com) and larger processes that perform heavier workloads usually end up on [AWS OpsWorks](http://aws.amazon.com/opsworks).
+
+Assets, including artwork images, are served from [Amazon S3](http://aws.amazon.com/s3/) via the [CloudFront CDN](http://aws.amazon.com/cloudfront).
+
+We send hundreds of thousands of e-mails via [SendGrid](http://sendgrid.com/) and [Mandrill](https://mandrill.com) and use [MailChimp](http://mailchimp.com) for manual campaigns. Our systems are monitored by a combination of [New Relic](http://newrelic.com/) and [Pingdom](https://www.pingdom.com). All of this is built, tested and deployed with [Jenkins](http://jenkins-ci.org), [Semaphore](https://semaphoreci.com), and [Travis-CI](https://travis-ci.org).
```

Comment:
I am adding a bunch of things related to this.

I threw Intercop into standardizing across CMSes:

_Artwork, artist and other metadata is uploaded by our partners via a home-grown content-management system built entirely from scratch in 2014 on Ruby-on-Rails. This was a deliberate "boring" choice that continues to serve us well. We have adopted a common model for admin-type apps with a shared component library and a reusable UI, all implemented as Rails engines. Using these components we are able to quickly and easily compose applications to manage fairs or auctions. We standardized on related services as well - for example, our customers interact with us via [Intercom](https://www.intercom.io)._

and separated Analytics

_We use [Statsd](https://github.com/etsy/statsd) and [Graphite](http://graphite.wikidot.com) for tracking high volume events. We recently began doing analytics with [Snowplow](https://github.com/snowplow/snowplow) and data warehousing in [AWS Redshift](http://aws.amazon.com/redshift), slowly moving away from [Chartio](https://chartio.com) and [Keen](https://keen.io). We store user actions via [Segment](https://segment.com) with [Google Analytics](http://www.google.com/analytics) and [MixPanel](https://mixpanel.com). It's fairly common to find a non-Engineer at Artsy in a Rails console directly querying data._